atlascope/core/models/pin.py

```diff
@@ -48,12 +42,116 @@ class Meta:
         ]
 
 
+class NotePin(Pin):
+    note = models.TextField(max_length=1000)
+
+
+class DatasetPin(Pin):
+    description = models.TextField(max_length=1000, blank=True)
+    child = models.ForeignKey(
+        'Dataset',
+        on_delete=models.CASCADE,
+        null=True,
+        related_name='locations',
+    )
+
+
+
+class NotePinSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = NotePin
+        fields = '__all__'
+
+
+class DatasetPinSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = DatasetPin
+        fields = '__all__'
+
+
 class PinSerializer(serializers.ModelSerializer):
+    def to_representation(self, instance):
```

Comment:
We now override `to_representation` in the Pin serializer. This is fine for serializing objects to JSON, and I expect this suffices for any `GET` endpoint. However, if we want to use this serializer to create objects e.g. from a `PUT` request, we'll also need to override `to_internal_value`.

`django-rest-polymorphic`'s `PolymorphicSerializer` is a subclass of DRF `Serializer` with all of this functionality build in already.

Reply:
I have some serious doubts about this approach. I mentioned it in the other PR, but I'm not seeing how `django-polymorphic` adds more debt than an approach like this, where you are manually adding code clauses for each subtype. But as I said in that thread, I think this deserves a live discussion.